src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/PackageSourceTelemetry.cs

```diff
@@ -222,11 +229,22 @@ internal static async Task<TelemetryEvent> ToTelemetryAsync(Data data, SourceRep
             TelemetryEvent telemetry;
             lock (data._lock)
             {
+                bool areNamespacesEnabled = packageNamespacesConfiguration?.AreNamespacesEnabled ?? false;
```

Comment:
`vs/nuget/packagesourcediagnostics`  emitted.

Reply:
@nkolev92 
Maybe I'll make follow up PR this event is emitted for install/update too. Is it ok?

Reply:
> @nkolev92
> Maybe I'll make follow up PR this event is emitted for install/update too. Is it ok?

There might be a different event. 

@zivkan can provide guidance here.

Reply:
I know there's some package source diagnostics in PM UI (although it has a bug where it only records telemetry for the initial search, but not all the requests to get deprecation/vulnerability pages, and is probably missing requests for getting the count of packages with updates).

Ideally yes, we should be recording telemetry for package install/uninstall. I kind of thought that it should be already, but it's entirely possible it's another bug in the implementation.